archunit/src/main/java/com/tngtech/archunit/core/importer/JavaClassProcessor.java

```diff
@@ -141,6 +142,13 @@ private boolean importAborted() {
         return javaClassBuilder == null;
     }
 
+    @Deprecated
+    @Override
+    public RecordComponentVisitor visitRecordComponentExperimental(int access, String name, String descriptor, String signature) {
```

Comment:
With ASM 8 (released [lately](https://asm.ow2.io/versions.html)), this needs to be:
```
public RecordComponentVisitor visitRecordComponent(String name, String descriptor, String signature) {
```

Reply:
Thanks for pointing out that ASM 8 got released, @hankem! I updated this PR.